### PR TITLE
ADO.NET integration

### PR DESCRIPTION
**Pull request by lucaspimentel**
Add a new integration that should cover all ADO.NET providers. 

Tested with:
- SQL Server, `System.Data.SqlClient` (replaces current integration)
- Postgres, `Npgsql`

Other ADO.NET providers (not tested yet):
- MySQL
- Oracle
- SQLite
- SQL Server Compact, `System.Data.SqlServerCe`
- OLE DB
- ODBC
- and more!

### CODE REVIEW

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/SqlServerTests.cs [hunk 1]
```diff
@@ -30,8 +30,8 @@ public void SubmitsTraces()
                 Assert.True(spans.Count > 0, "expected at least one span");
                 foreach (var span in spans)
                 {
-                    Assert.Equal(SqlServerIntegration.OperationName, span.Name);
-                    Assert.Equal($"Samples.SqlServer-{SqlServerIntegration.ServiceName}", span.Service);
+                    Assert.Equal("sql-server.query", span.Name);
```
Comment (bmermet) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/SqlServerTests.cs]: Why do we not also have integration tests for postgres?

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        /// <summary>
+        /// Wrapper method that instruments <see cref="System.Data.Common.DbCommand.ExecuteDbDataReader"/>.
+        /// </summary>
+        /// <param name="this">The <see cref="DbCommand"/> that is references by the "this" pointer in the instrumented method.</param>
+        /// <param name="behavior">A value from <see cref="CommandBehavior"/>.</param>
+        /// <returns>The value returned by the instrumented method.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Data", // .NET Framework
+            TargetType = "System.Data.Common.DbCommand",
+            TargetSignature = "20 01")]
```
Comment (bmermet) [inline src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs]: Why do we have to explicitly set the signature for this integration? Coulnd't it be computed automatically by the json generator?

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 3]
```diff
@@ -271,6 +271,12 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
     return S_OK;
   }
 
+  /*
```
Comment (bmermet) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Remove if not useful

File: samples/Samples.Npgsql/Program.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            Console.WriteLine($"Profiler attached: {Instrumentation.ProfilerAttached}");
+            Console.WriteLine($"Platform: {(Environment.Is64BitProcess ? "x64" : "x32")}");
+            Console.WriteLine();
+
+            using (var conn = new NpgsqlConnection("Host=localhost;Username=postgres;Password=postgres;Database=postgres"))
+            {
+                conn.Open();
+
+                /*
```
Comment (bmermet) [inline samples/Samples.Npgsql/Program.cs]: Shouldn't this be uncommented?

File: samples/Samples.Npgsql/Program.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                            int count = reader.GetValues(values);
+                            Console.WriteLine(string.Join(", ", values.Take(count)));
+                            rows++;
+                        }
+                    }
+
+                    Console.WriteLine($"{rows:N0} rows returned from async query.");
+                }
+
+                /*
```
Comment (bmermet) [inline samples/Samples.Npgsql/Program.cs]: Shouldn't this be uncommented?

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs (hunk 2 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs]: This is left over from when we instrumented `ExecuteReader()`. We needed the signature to distinguish between overloads. It looks like `ExecuteDbDataReader()` doesn't have overloads so we can probably drop this signature. I'll test that.

The json generator does figure out the signature for the wrapper method `AdoNetIntegration.ExecuteDbDataReader()`, so we don't need to include that here.

File: samples/Samples.Npgsql/Program.cs (hunk 5 above)
Comment (lucaspimentel) [reply samples/Samples.Npgsql/Program.cs]: Oops. I've been running this test app manually, so I comment/uncomment as needed.

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/SqlServerTests.cs (hunk 1 above)
Comment (lucaspimentel) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/SqlServerTests.cs]: The short answer is "time." Postgres support is time-sensitive. I tested Postgres manually and the SQL Server tests cover ADO.NET as well. I'm adding more automated integration tests for different ADO.NET providers in a separate PR.

File: src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs (hunk 2 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs]: The signatures were not needed here, so I removed them.

**Comment by hemant-learningforever** [comment]
@lucaspimentel  Could you please explain how are you getting this signature for wrapper type?

**Comment by lucaspimentel** [comment]
@hemant-learningforever: This work on this PR is finished and merged. Please ask questions by creating a new [issue](https://github.com/DataDog/dd-trace-dotnet/issues).

**Pull request merged**